crates/oxc_ast/src/visit/visit_mut.rs

```diff
@@ -9,823 +9,948 @@ use crate::{ast::*, AstType};
 #[allow(clippy::wildcard_imports)]
 use self::walk_mut::*;
 
+pub trait VisitResult {
```

Comment:
I'm not sure if this pattern would work, We can have much better control if we use default and many. Where we want to visit a list we need a way to construct a single result from all of those results in the list.

This way we can leave those untouched in implementations other than Transformer, It is the same for BasicVisitResult, if instead of using a concrete type we just implement `VisitResult` for unit (`()`) we can have methods that don't need to explicitly return.